events/janus_gelfevh.c

```diff
@@ -589,25 +590,28 @@ static void *janus_gelfevh_handler(void *data) {
 			json_t *microtimestamp = json_object_get(event, "timestamp");
 			if(microtimestamp && json_is_integer(microtimestamp)) {
 				double created_timestamp = (double)json_integer_value(microtimestamp) / 1000000;
-				json_object_set(output, "timestamp", json_real(created_timestamp));
+				json_object_set_new(output, "timestamp", json_real(created_timestamp));
 			} else {
-				json_object_set(output, "timestamp", json_real(janus_get_real_time()));
+				json_object_set_new(output, "timestamp", json_real(janus_get_real_time()));
 			}
 			json_object_set(output, "host", json_object_get(event, "emitter"));
-			json_object_set(output, "version", json_string("1.1"));
+			json_object_set_new(output, "version", json_string("1.1"));
 			json_object_set(output, "level", json_object_get(event, "type"));
-			json_object_set(output, "short_message", json_string(short_message));
+			json_object_set_new(output, "short_message", json_string(short_message));
 			json_object_set(output, "full_message", event);
 
-			if(janus_gelfevh_send(json_dumps(output, json_format)) < 0) {
-				JANUS_LOG(LOG_WARN, "Couldn't send event to GELF, reconnect?, or event was null: %s\n",
-					json_dumps(output, json_format));
+			char *message = json_dumps(output, json_format);
+			if(janus_gelfevh_send(message) < 0) {
+				JANUS_LOG(LOG_WARN, "Couldn't send event to GELF, reconnect?, or event was null: %s\n", message);
 			}
 			json_decref(output);
+			free(message);
```

Comment:
Apologies for the late correction, but this should also have a check on whether `message` is `NULL`, as while `g_free` checks this automatically, I don't think `free` does. I'll have to check if we do that in other handles or transports, but if we don't we should.

Reply:
`free(NULL)` is safe and well-defined:
> The free function causes the space pointed to by ptr to be deallocated, that is, made available for further allocation. If ptr is a null pointer, no action occurs. [1]

[1] http://www.open-std.org/JTC1/SC22/wg14/www/docs/n1124.pdf

Reply:
I'm having deja-vu, which means I probably said the same silly thing in the past and you corrected me back then as well :smile:  Thanks for the clarification!

Reply:
hope this should be good to avoid double freeing , thanks